src/commands/doctor.rs

```diff
@@ -0,0 +1,115 @@
+use std::{fmt::Debug, path::PathBuf};
+
+use anyhow::Result;
+use clap::Parser;
+use dialoguer::{console::Emoji, Confirm, Select};
+use futures::FutureExt;
+use spin_doctor::{Diagnosis, DryRunNotSupported};
+
+#[derive(Parser, Debug)]
+#[clap(hide = true, about = "Detect and fix problems with Spin applications")]
+pub struct DoctorCommand {
+    #[clap(short = 'f', long, default_value = "spin.toml")]
+    file: PathBuf,
```

Comment:
Petition to allow a directory here as we do in `spin up`.  (I know this is not consistent through `build` and `deploy` yet, so no biggie if you prefer to punt on it - I can do a "-f dir everywhere" pass later.  ~Or I could do it now and make YOU have the merge conflicts.~  uh not that I would ever do a thing like that)

Reply:
:+1: Yeah lets tackle this in a separate PR. This is already turning into a giga-PR.

Reply:
For future reference I don't think `spin watch` supports directories for `-f`. I think this flag/functionality is perfect for the common crate.

Reply:
> perfect for the common crate

Yes, except that I believe all the users would be in the root `spin-cli` crate, in which case it should just go there :slightly_smiling_face:

Reply:
Not if/when we move `spin watch` to its own crate.

Reply:
Yeah, this is one of those things that should be resolved at the UI level, and everything below there should just not have to think about it.  I'll pick this up (and no worries @calebschoepp `spin watch` is ~in my sights~ uh ~on my radar~ uh appended to my to-do list).